src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/BitbucketCloudEndpoint.java

```diff
@@ -77,10 +77,12 @@ public BitbucketCloudEndpoint(boolean manageHooks, @CheckForNull String credenti
      * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
      * @param credentialsId The {@link StandardCredentials#getId()} of the credentials to use for
      *                      auto-management of hooks.
-     */
+     * @param bitbucketJenkinsRootUrl The custom (or empty for global setting) Jenkins Root URL
+     *                      auto-management of hooks.
+      */
     @DataBoundConstructor
-    public BitbucketCloudEndpoint(boolean enableCache, int teamCacheDuration, int repositoriesCacheDuration, boolean manageHooks, @CheckForNull String credentialsId) {
-        super(manageHooks, credentialsId);
+    public BitbucketCloudEndpoint(boolean enableCache, int teamCacheDuration, int repositoriesCacheDuration, boolean manageHooks, @CheckForNull String credentialsId, @CheckForNull String bitbucketJenkinsRootUrl) {
```

Comment:
I would prefer that this was not added to the `@DataboundConstructor`.

Instead, use a `@DataboundSetter` since the value is not mandatory. This also removes the need for passing empty strings around.

Reply:
Why shouldn't that field be added to the constructor? It is a common property of all descendants of the abstract class, including our Cloud and Server configs, so empty or not - it is sort of required and always set... But maybe I just misunderstand the magic behind this instantiation.

Reply:
Also, looking now at where to make the change, I remembered that this abstract class did not provide any setters for configurable values, so I was not sure it is possible/proper to add any. Instead, all the common variables (credentials and Manage boolean) were set via constructor.